src/Xamarin.Android.Build.Tasks/Utilities/MavenExtensions.cs

```diff
@@ -130,45 +131,57 @@ public static bool TryParseJavaArtifacts (this ITaskItem task, string type, Task
 	}
 
 	// Returns artifact output path
-	public static async Task<string?> DownloadPayload (CachedMavenRepository repository, Artifact artifact, string cacheDir, TaskLoggingHelper log, CancellationToken cancellationToken)
+	public static async Task<string?> DownloadPayload (CachedMavenRepository repository, Artifact artifact, string cacheDir, string? mavenOverrideFilename, TaskLoggingHelper log, CancellationToken cancellationToken)
 	{
 		var output_directory = Path.Combine (cacheDir, repository.Name, artifact.GroupId, artifact.Id, artifact.Version);
 
 		Directory.CreateDirectory (output_directory);
 
-		var filename = $"{artifact.Id}-{artifact.Version}";
-		var jar_filename = Path.Combine (output_directory, Path.Combine ($"{filename}.jar"));
-		var aar_filename = Path.Combine (output_directory, Path.Combine ($"{filename}.aar"));
+		var files_to_check = new List<string> ();
+
+		if (mavenOverrideFilename.HasValue ()) {
+			files_to_check.Add (Path.Combine (output_directory, mavenOverrideFilename));
+		} else {
+			files_to_check.Add (Path.Combine (output_directory, $"{artifact.Id}-{artifact.Version}.jar"));
+			files_to_check.Add (Path.Combine (output_directory, $"{artifact.Id}-{artifact.Version}.aar"));
```

Comment:
We don't have to do this here, but is there some manifest file that we can download to know the exact file name? It seems odd they can choose random names. How does Java tooling know which file to download?

Reply:
That is a good question. There doesn't appear to be anything in the `.pom` that gives the file name: https://repo1.maven.org/maven2/com/facebook/react/react-android/0.76.0/react-android-0.76.0.pom

It looks like React has its own Gradle plugin you need to use, perhaps it knows how to resolve their files?